src/main/java/net/fabricmc/loom/LoomGradleExtension.java

```diff
@@ -425,4 +432,113 @@ public synchronized File getNextMixinMappings() {
 	public Set<File> getAllMixinMappings() {
 		return Collections.unmodifiableSet(mixinMappings);
 	}
+
+	public void runs(Closure<?> conf) {
+		runs.configure(conf);
+	}
+
+	public NamedDomainObjectContainer<RunConfigSettings> getRuns() {
+		return runs;
+	}
+
+	public static class RunConfigSettings implements Named {
+		private final List<String> vmArgs = new ArrayList<>();
+		private final List<String> programArgs = new ArrayList<>();
+		private String mode;
+		private String name;
+		private Boolean client;
+		private final String baseName;
+
+		public RunConfigSettings(String baseName) {
+			this.baseName = baseName;
+			setMode(baseName);
+		}
+
+		@Override
+		public String getName() {
+			return baseName;
+		}
+
+		public List<String> getVmArgs() {
+			return vmArgs;
+		}
+
+		public List<String> getProgramArgs() {
+			return programArgs;
+		}
+
+		public String getMode() {
+			return mode;
+		}
+
+		public void setMode(String mode) {
+			this.mode = mode;
+		}
+
+		public String getConfigName() {
+			return name;
+		}
+
+		public void setConfigName(String name) {
+			this.name = name;
+		}
+
+		public boolean isClient() {
+			String m = mode != null ? mode : baseName;
+			return client != null ? client // Do not confuse users: detect client mode unless client mode is explicitly defined
+					: m.toLowerCase().contains("client");
+		}
+
+		public void setClient(Boolean client) {
+			this.client = client;
+		}
+
+		public void mode(String mode) {
+			setMode(mode);
+		}
+
+		public void name(String name) {
+			setConfigName(name);
+		}
+
+		public void client(boolean client) {
+			setClient(client);
+		}
+
+		public void server(boolean server) {
+			setClient(!server);
+		}
+
+		public void vmArg(String arg) {
+			vmArgs.add(arg);
+		}
+
+		public void vmArgs(String... args) {
+			vmArgs.addAll(Arrays.asList(args));
+		}
+
+		public void vmArgs(Collection<String> args) {
+			vmArgs.addAll(args);
+		}
+
+		public void property(String name, String value) {
```

Comment:
Would be nice if there was an overload of this which just passed the name so we can `-Dname`

Reply:
Done